src/sqlparser/src/parser.rs

```diff
@@ -404,6 +404,9 @@ impl Parser {
                 Keyword::ARRAY => Ok(Expr::Array(
                     self.parse_token_wrapped_exprs(&Token::LBracket, &Token::RBracket)?,
                 )),
+                k if keywords::RESERVED_FOR_COLUMN_OR_TABLE_NAME.contains(&k) => {
+                    parser_err!(format!("syntax error at or near \"{w}\""))
+                }
```

Comment:
CREATE is not affected

```sql
dev=> create table order(x int, order int);
CREATE_TABLE
dev=> select * from order;
 x | order
---+-------
(0 rows)

dev=> select order.x from order;
ERROR:  QueryError: sql parser error: syntax error at or near "order"
dev=>
```

Reply:
Updated to reserve for column in CREATE.

Note that we still allow these words as table names in some cases:
```
dev=> create table order (v1 int);
CREATE_TABLE
dev=> select v1 from order;
 v1 
----
(0 rows)
```

Fixing this requires a bigger effort. As PostgreSQL does allow it when qualified:
```
test=# create table order (v1 int);
ERROR:  syntax error at or near "order"
LINE 1: create table order (v1 int);
                     ^
test=# create table public.order (v1 int);
CREATE TABLE
```